### PR TITLE
Add `descriptions.tsv` to the schema

### PR DESCRIPTION
**Pull request by tsalo**
Closes none, but continues from #1613.

I didn't figure out how to render the filenames in a macro. I think allowing `descriptions.tsv` files at the subject or session level introduced more complexity than it was worth, unfortunately.

I also haven't figured out how to add meaningful rules about the file to the schema. Specifically, we probably want to raise a warning if any desc values described in the file aren't in the dataset, or vice versa.

Changes proposed:

- Add the `descriptions.tsv` file to the schema.

### CODE REVIEW

File: src/schema/objects/columns.yaml [hunk 1]
```diff
@@ -122,6 +129,12 @@ description__optode:
   description: |
     Free-form text description of the optode, or other information of interest.
   type: string
+description__descriptions:
```
Comment (tsalo) [inline src/schema/objects/columns.yaml:132]: If we choose to implement other entity definition TSVs (see #1706), I wonder if we'd want to use the same column name ("description") for the value definitions or if we'd want to use the full entity names (e.g., `acq_id` and `acquisition`). I think `acq_id` and `description` makes more sense TBH, in which case maybe the identifier here should be `description__entity`.

**Comment by tsalo** [comment]
Switching to a draft until I figure out how to render the filename template.

**Comment by tsalo** [comment]
I've given up on rendering the template with a macro, but I'll bring up adding checks for the file in the schema meeting today.

Presumably we would want rules to check if all `desc-<label>` entities are reflected in the file and that no definitions are included that _aren't_ present in the dataset. Both should probably be warnings, rather than errors.

File: src/derivatives/common-data-types.md [hunk 2]
```diff
@@ -256,24 +256,33 @@ static volume, a `RepetitionTime` property would no longer be relevant).
 
 ## descriptions.tsv
 
+Template:
+
+```Text
+sub-<label>/
+    [ses-<label>/]
+        sub-<label>[_ses-<label>]_descriptions.tsv
+        sub-<label>[_ses-<label>]_descriptions.json
```
Comment (effigies) [inline src/derivatives/common-data-types.md:265]:
```suggestion
[sub-<label>/]
    [ses-<label>/]
        [sub-<label>][_ses-<label>]_descriptions.tsv
        [sub-<label>][_ses-<label>]_descriptions.json
```

File: src/schema/objects/columns.yaml (hunk 1 above)
Comment (tsalo) [reply src/schema/objects/columns.yaml:140]: Per chat with @effigies, `description__entity` makes more sense. We can customize the definitions as they appear in the specification text with the macros.

File: src/derivatives/common-data-types.md (hunk 2 above)
Comment (tsalo) [reply src/derivatives/common-data-types.md:265]: Oh yeah good point!

**Review by effigies — APPROVED** [review]
LGTM.

File: src/schema/objects/files.yaml [hunk 3]
```diff
@@ -126,3 +126,8 @@ stimuli:
     A directory to store any stimulus files used during an experiment.
     See the [relevant section](SPEC_ROOT/modality-specific-files/task-events.md#stimuli-directory)
     for more information.
+descriptions:
+  display_name: Descriptions Definitions
+  file_type: regular
+  description: |
+    A TSV file describing labels found for the `desc` entity in a Derivatives dataset.
```
Comment (tsalo) [inline src/schema/objects/files.yaml:133]: This actually might conflict with the suffix definition later. Should I remove it?

Comment (tsalo) [reply src/schema/objects/files.yaml]: Removing it.

**Comment by tsalo** [comment]
I will open an issue about my two pain points:

- Filename template building isn't working.
- Rules/checks not implemented in schema.

**Review by sappelhoff — APPROVED** [review]
approving for merge as soon as "good enough" and to then cover the pain points in a follow-up (or several).

**Comment by Remi-Gau** [comment]
@tsalo 
Can't remember if there is a reason this was not merged?

**Comment by tsalo** [comment]
No reason- I didn't realize I had two approvals. I'll update from main and then merge once CI passes.

**Pull request merged**